src/antlr/index.ts

```diff
@@ -309,7 +27,11 @@ function parse(sql: string, options?: LineageParserOptions | CompletionParserOpt
   parser.buildParseTree = true;
   parser.removeErrorListeners();
 
-  return new SqlCompletionParseTree(parser.statement(), defaultCursor);
+  if (options !== undefined && isCompletionOptions(options)) {
+    return new SqlCompletionParseTree(parser.statement(), defaultCursor);
+  } else {
+    return new SqlLineageParseTree(parser.statement());
+  }
```

Comment:
@megaflo I changed this to return either the completion or the lineage parse tree - in line with the type overloads.

Reply:
Lets separate parse for lineage and parse for completion fully. These overloads just make things more difficult to understand.

Reply:
@pavelgubin We (@megaflo + me) do like the external interface 

```typescript
function parse(sql: string, options: CompletionParserOptions): SqlCompletionParseTree;
function parse(sql: string, options?: LineageParserOptions): SqlLineageParseTree;
```

It is quite simple to use - specify a cursor and you can use completion functionality, do not specify a cursor and you get "only" lineage functionality. I like the idea of a single entry point to the antlr functionality.

Do you have a strong preference to change this?

Reply:
I'm ok if you both are ok, for me still a bit strange that result is decided by the field inside options.